src/app/clusters/commissioner-control-server/commissioner-control-server.cpp

```diff
@@ -0,0 +1,251 @@
+
+    auto sourceNodeId = getNodeId(commandObj);
+    auto fabricIndex  = commandObj->GetAccessingFabricIndex();
+    auto requestId    = commandData.requestId;
+    auto vendorId     = commandData.vendorId;
+    auto productId    = commandData.productId;
+    auto & label      = commandData.label;
+
+    // Create a CommissioningApprovalRequest struct and populate it with the command data
+    Clusters::CommissionerControl::CommissioningApprovalRequest request = { .requestId    = requestId,
+                                                                            .vendorId     = vendorId,
+                                                                            .productId    = productId,
+                                                                            .clientNodeId = sourceNodeId,
+                                                                            .fabricIndex  = fabricIndex,
+                                                                            .label        = label };
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(sourceNodeId != kUndefinedNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Handle commissioning approval request
+    err = delegate->HandleCommissioningApprovalRequest(request);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "emberAfCommissionerControlClusterRequestCommissioningApprovalCallback error: %s", err.AsString());
+        status = Status::Failure;
+    }
+
+    commandObj->AddStatus(commandPath, status);
+    return true;
+}
+
+bool emberAfCommissionerControlClusterCommissionNodeCallback(
+    app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
+    const Clusters::CommissionerControl::Commands::CommissionNode::DecodableType & commandData)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    Status status  = Status::Success;
+
+    ChipLogProgress(Zcl, "Received command to commission node");
+
+    auto sourceNodeId = getNodeId(commandObj);
+    auto requestId    = commandData.requestId;
+
+    auto info       = std::make_unique<Clusters::CommissionerControl::CommissionNodeInfo>();
+    info->ipAddress = commandData.ipAddress;
+    info->port      = commandData.port;
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(sourceNodeId != kUndefinedNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Handle commissioning approval request
+    err = delegate->ValidateCommissionNodeCommand(sourceNodeId, requestId, info->params);
+    VerifyOrExit(err == CHIP_NO_ERROR, err = CHIP_ERROR_INCORRECT_STATE);
```

Comment:
nit, what is the purpose of setting the error code to `err = CHIP_ERROR_INCORRECT_STATE`? I don't see it used 
```suggestion
    VerifyOrExit(err == CHIP_NO_ERROR);
```